src/features/bounds/model.ts

```diff
@@ -99,6 +100,22 @@ export function getAbsoluteBounds(element: SModelElement): Bounds {
     }
 }
 
+/**
+ * Determine whether the given element is visible in the current canvas.
+ */
+export function isVisible(element: Readonly<SModelElement>, context?: RenderingContext): boolean {
```

Comment:
Currently this function needs to be called explicitly in the view implementation, e.g. [circlegraph example](https://github.com/eclipse/sprotty/pull/182/files#diff-175df6bdd8169e4881c16da111261cde). Alternatively, we could do this check for all elements in ModelRenderer, or make it depend on an optional IView property. We cannot move it to IView because it's an interface. We don't have a common superclass for views.

Reply:
@JanKoehnlein what do you think? Shall we make the visibility check more prominent?

Reply:
Maybe. I remember a customer project where we had to manually implement some level-of-detail in the views depending on the zoom level, and it looks like this should be done in the same hook (as it is also based on absolute bounds). Combined with your viewport check this has a lot of potential to speed up big and deeply nested diagrams, so why not place it more prominently and make it default?

Alternatively, we could document it and refer to the examples.

Reply:
I tried moving the check to ModelRenderer. There are two problems with this:
 * It adds dependencies to `features/bounds/model` and `features/routing/model`.
 * For edges, we can only determine whether they are visible *after* we have computed their routing.
Therefore I added the visibility checks to the default views instead.

Reply:
Maybe that's a separate service/feature?